### PR TITLE
Fixes for inserting/removing storage items/swapping items between hands.

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR fixes the behavior where items inserted into or removed from storage would, for an instant, be moved to the ground before their actual destination. It fixes the same issue with swapping an item between hands.

There's a raft of checks that need to occur when performing this action, including checking for storage compatibility, checking for ability to hold items, checking for presence of hands, etc., and I'm skeptical that all of those checks were being performed in all cases, before or after this PR. I tried to make this fix as surgical as possible and worry about all the other shit later when we port atom storage.
## Why It's Good For The Game
This fixes bugs involving items catching on fire when moving them from/into storage on top of lava/falling into chasms when same, fixes some behavior with the upcoming #26637, and is just the correct thing to do programmatically.
## Testing

https://github.com/user-attachments/assets/e2417a24-07fa-41b3-b90f-ea2185dc0f91

<hr>

### Declaration

- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
fix: Items will no longer catch on fire when swapping them between hands, or inserting into or removing them from bags/boxes/belts etc. over lava.
fix: Items will no longer fall into chasms when swapping them between hands, or inserting into or removing them from bags/boxes/belts etc. over chasms.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/robot_inventory.dm [hunk 1]
```diff
@@ -239,7 +239,7 @@
 	while(slot_start != slot_num) //If we wrap around without finding any free slots, just give up.
 	return
 
-/mob/living/silicon/robot/unEquip(obj/item/I, force, silent = FALSE)
+/mob/living/silicon/robot/unEquip(obj/item/I, force = FALSE, silent = FALSE, atom/destination = null)
```
Comment (DGamerL) [inline code/modules/mob/living/silicon/robot/robot_inventory.dm:242]:
```suggestion
/mob/living/silicon/robot/unEquip(obj/item/I, force = FALSE, silent = FALSE, atom/destination)
```

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 2]
```diff
@@ -504,7 +504,7 @@
 		if(ITEM_SLOT_COLLAR)
 			add_collar(W)
 
-/mob/living/simple_animal/unEquip(obj/item/I, force, silent = FALSE)
+/mob/living/simple_animal/unEquip(obj/item/I, force = FALSE, silent = FALSE, atom/destination = null)
```
Comment (DGamerL) [inline code/modules/mob/living/simple_animal/simple_animal.dm:507]:
```suggestion
/mob/living/simple_animal/unEquip(obj/item/I, force = FALSE, silent = FALSE, atom/destination)
```

File: code/modules/mob/living/simple_animal/slime/slime_mob.dm [hunk 3]
```diff
@@ -266,7 +266,7 @@
 			return
 	return ..()
 
-/mob/living/simple_animal/slime/unEquip(obj/item/I, force, silent = FALSE)
+/mob/living/simple_animal/slime/unEquip(obj/item/I, force = FALSE, silent = FALSE, atom/destination = null)
```
Comment (DGamerL) [inline code/modules/mob/living/simple_animal/slime/slime_mob.dm:269]:
```suggestion
/mob/living/simple_animal/slime/unEquip(obj/item/I, force = FALSE, silent = FALSE, atom/destination)
```

File: code/modules/mob/inventory_procs.dm [hunk 4]
```diff
@@ -166,13 +193,20 @@
 
 	return TRUE
 
-/mob/proc/unEquip(obj/item/I, force, silent = FALSE) //Force overrides NODROP for things like wizarditis and admin undress.
+/// Unequip an item from the hand that the item is found in.
+/// `force` overrides NODROP for things like wizarditis and admin undress.
+/// `destination` allows for items to be unequipped directly into storage and should only be used for that.
+/// Horrid stop-gap until we get atom storage or something.
+/mob/proc/unEquip(obj/item/I, force = FALSE, silent = FALSE, atom/destination = null)
```
Comment (DGamerL) [inline code/modules/mob/inventory_procs.dm:200]:
```suggestion
/mob/proc/unEquip(obj/item/I, force = FALSE, silent = FALSE, atom/destination)
```

**Review by DGamerL — COMMENTED** [review]
Enjoy a phone review!

**Review by DGamerL — APPROVED** [review]
LGTM!

**Review by Burzah — APPROVED** [review]

**Pull request merged**